gcc or binutils seems to have been broken recently

My daily build two nights ago worked.  A fresh clone of github.com/llnl/spack late yesterday afternoon failed to build, as did my build last night.

I'm building gcc@5.4.0 with the CentOS system compiler, gcc@4.8.[35].

The compiler appears to be failing.

Here's a bit from a `config.log` file.  The interesting bit is `n--eh-frame-hdr`, it looks as if there might be a stray 'n' somewhere that's getting pasted onto the front of the `--eh-frame-hdr` switch:

```
configure:3851: /scratch/hartzelg/working-dir/workspace/daily-build/spack/lib/spack/env/gcc/gcc    conftest.c  >&5
/blah/apps/spack/v0.0.10/opt/spack/linux-centos7-x86_64/gcc-4.8.5/binutils-2.28-66ufycihw5nglnjuy4xf6hqpwrru6bx4/bin/ld: cannot find n--eh-frame-hdr: No
 such file or directory
collect2: error: ld returned 1 exit status
```

I've been fishing for the change, but perhaps it will ring a bell.

--- a/var/spack/repos/builtin/packages/gcc/package.py
+++ b/var/spack/repos/builtin/packages/gcc/package.py
@@ -250,6 +250,6 @@ class Gcc(AutotoolsPackage):
             for line in lines:
                 out.write(line + '\n')
                 if line.startswith('*link:'):
-                    out.write('-rpath {0}:{1} \\\n'.format(
+                    out.write('-rpath {0}:{1} '.format(
                               self.prefix.lib, self.prefix.lib64))
         set_install_permissions(specs_file)

--- a/var/spack/repos/builtin/packages/gcc/package.py
+++ b/var/spack/repos/builtin/packages/gcc/package.py
@@ -250,6 +250,6 @@ class Gcc(AutotoolsPackage):
             for line in lines:
                 out.write(line + '\n')
                 if line.startswith('*link:'):
-                    out.write(r'-rpath {0}:{1} \n'.format(
+                    out.write('-rpath {0}:{1} \\\n'.format(
                               self.prefix.lib, self.prefix.lib64))
         set_install_permissions(specs_file)

